src/sagemaker/mxnet/estimator.py

```diff
@@ -110,34 +108,31 @@ def __init__(
             :class:`~sagemaker.estimator.Framework` and
             :class:`~sagemaker.estimator.EstimatorBase`.
         """
-        if framework_version is None:
+        self.framework_version = framework_version
+        self.py_version = py_version
```

Comment:
updated thoughts after your conversation this morning and looking at the code in `attach`: I think (at least for this pass) let's let both still default to `None`, but validate that if they are `None` then `image_name is not None` and raise an exception otherwise. (and update the docstrings to reflect this.)

Reply:
agreed. will update